### PR TITLE
Resnet18 best results & iterative training + async training

### PR DESCRIPTION
**Pull request by rayaq-siddiqui**

### CODE REVIEW

File: main_distributed_iteration_pub_sub.py
Comment (rayaq-siddiqui) [inline main_distributed_iteration_pub_sub.py:1]: @Joon1221 This is the main script for these changes

File: experiments/iteration_based/main_distributed_iteration_based.py
Comment (rayaq-siddiqui) [inline experiments/iteration_based/main_distributed_iteration_based.py:1]: not important

File: experiments/iteration_based/worker_distributed_iteration_based.py
Comment (rayaq-siddiqui) [inline experiments/iteration_based/worker_distributed_iteration_based.py:1]: not important

**Comment by rayaq-siddiqui** [comment]
Note: the wandb work is also in this PR. It is included in the experimentation folder

**Comment by Joon1221** [comment]
I'm a little unsure how these changes actually reduce the time it takes for the processes start, but aside from that, everything LGTM

**Pull request merged**